command/rootchain/deploy/deploy.go

```diff
@@ -546,43 +573,72 @@ func deployContracts(outputter command.OutputFormatter, client *jsonrpc.Client,
 	}
 
 	if err := g.Wait(); err != nil {
-		messageResult := helper.MessageResult{
-			Message: "[ROOTCHAIN - DEPLOY] Successfully deployed the following contracts\n"}
-
-		for i, result := range results {
-			if result != nil {
-				// In case an error happened, some of the indices may not be populated.
-				// Filter those out.
-				commandResults[i] = result
-			}
+		return collectResultsOnError(results), err
+	}
+
+	proxyAdmin := types.StringToAddress(params.proxyContractsAdmin)
+
+	g, ctx = errgroup.WithContext(cmdCtx)
+
+	for _, contract := range allContracts {
```

Comment:
Just curious. Is deploying proxies separately done because of an optimization, or some other reason?

Reply:
Yes- proxy contract constructor accepts implementation contract address as argument, so we need to deploy implementation contracts first.

Reply:
I was thinking can we deploy a contract and its proxy in single routine. I don't know what are the differences in execution time if we deploy proxy in the same routine as contract, and separately. If the difference is significant then leave it like this, if it is not, then it will be simpler to just deploy it in the same routine as contract.

Reply:
Ah, I didn't understand your question properly. Yes we could do that. I guess the difference wouldn't be noticeable, so I will do it.

Reply:
Fixed 63b8373897db3bd9a21f2245d0dfe9aab726f29e